### PR TITLE
Fixed issue when contract deployment burns previous balance

### PR DESCRIPTION
**Pull request by Vovchyk**
Contract addresses are deterministic. So it's possible to calculate an address of a contract before actual deployment (for some particular Externally Owned Account and nonce).

This pull request addresses the issue when some funds are transferred in advance to the contract address which is not deployed yet. And then after deployment the balance becomes zero.

Note: The issue is only relevant to the case when a contract is created directly, not via `CREATE` or `CREATE2` opcodes.

### CODE REVIEW

**Comment by Vovchyk** [comment]
pipeline: run

File: rskj-core/src/main/java/org/ethereum/core/TransactionExecutor.java [hunk 1]
```diff
@@ -369,7 +369,15 @@ private void call() {
...
-        cacheTrack.createAccount(newContractAddress); // pre-created
+
+        // Keep prior balance of a contract address (if any)
+        if (cacheTrack.isExist(newContractAddress)) {
+            Coin oldBalance = cacheTrack.getBalance(newContractAddress);
+            cacheTrack.createAccount(newContractAddress);
+            cacheTrack.addBalance(newContractAddress, oldBalance);
+        } else {
+            cacheTrack.createAccount(newContractAddress);
+        }
```
Comment (ajlopezrsk) [inline rskj-core/src/main/java/org/ethereum/core/TransactionExecutor.java:380]: Why not simply NOT create account?

Comment (Vovchyk) [reply rskj-core/src/main/java/org/ethereum/core/TransactionExecutor.java:380]: I just implemented it the same way as it's done for `create` and `create2` opcodes - https://github.com/rsksmart/rskj/blob/master/rskj-core/src/main/java/org/ethereum/vm/program/Program.java#L545

`Repository.createAccount` fully recreates an account from scratch. Do you thing we should simplify this logic and do nothing for existing accounts?

**Review by patogallaiovlabs — DISMISSED** [review]
LGTM

**Review by patogallaiovlabs — APPROVED** [review]

**Review by patogallaiovlabs — APPROVED** [review]

**Review by patogallaiovlabs — APPROVED** [review]
LGTM

**Pull request merged**